### DISCUSSION TITLE
Generic socket error

### DISCUSSION THREAD

**Discussion by lucadici** [cesanta/mongoose#1867] [Ask questions, get answers] [answered]
- My goal is: understand how big can be the buffer sent via mg_send
- My actions were: running the code below:
 
```
char overMaxData[4*1024*1024]; //large buffer
mg_http_reply(c, 100, "","%s", overMaxData); //or equivalently, mg_send(c, (void*)overMaxData, sizeof(overMaxData)-1);
```
- My expectation was: to receive a more descriptive error message
- The result I saw: 

`241e3c2 1 mongoose.c:404:mg_error       2 0x6 socket error
`
this comes from:

```
  for (int i = 0; i < n; i++) {
    struct mg_connection *c = (struct mg_connection *) evs[i].data.ptr; 
    if (evs[i].events & **EPOLLERR**) {
      **mg_error(c, "socket error");**
```

- My question is: what is the cause of EPOLLERR? This cannot be understood from the log message.

Environment

- mongoose version: _7.8_
- Compiler/IDE and SDK: gcc 9.4.0
- Target hardware/board: Ubuntu virtual machine, Linux ubuntu2004 5.15.0-52-generic
- Connectivity chip/module: localhost

**Comment by scaprile** [comment, marked as answer]
If you are sending on a TCP connection, Mongoose will try resize the TCP buffer to hold your data and then send it to the socket interface in your OS.
The [http-server](https://github.com/cesanta/mongoose/tree/master/examples/http-server) example as is serves 4MB in RHEL. I even compiled with MG_ENABLE_EPOLL (which seems to be what you are using) and it worked OK. There is no limit per se.
Please enable debug or verbose messages and track why your OS is rejecting that transfer.
`mg_log_set( log level )`
https://github.com/cesanta/mongoose/blob/d958ec2ac877b94520c7c300932af251a254af1b/mongoose.h#L810-L813
